### PR TITLE
fix(function): fix incorrect return datatype of function if

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

## Changelog

 
- Bug Fix
 

## Related Issues

Part of #5959  #5972

### CODE REVIEW

**Review by leiysky — APPROVED** [review]

**Comment by leiysky** [comment]
I'm pretty sure that my previous PR #5119 is way blocked by this issue...

**Comment by xudong963** [comment]
As expected, they failed :D
image

File: common/datablocks/tests/it/kernels/data_block_sort.rs [hunk 1]
```diff
@@ -185,7 +185,7 @@ fn test_data_block_sort() -> Result<()> {
             asc: true,
             nulls_first: false,
         }];
-        println!("raw={:?}", raw);
+        // println!("raw={:?}", raw);
```
Comment (BohuTANG) [inline common/datablocks/tests/it/kernels/data_block_sort.rs:188]: I guess this code is no used anymore.

File: common/datablocks/src/data_block.rs [hunk 2]
```diff
@@ -34,6 +34,11 @@ pub struct DataBlock {
 impl DataBlock {
     #[inline]
     pub fn create(schema: DataSchemaRef, columns: Vec<ColumnRef>) -> Self {
+        debug_assert!(schema.fields().iter().zip(columns.iter()).all(|(f, c)| f
```
Comment (Xuanwo) [inline common/datablocks/src/data_block.rs:37]: Nice!

**Pull request merged**